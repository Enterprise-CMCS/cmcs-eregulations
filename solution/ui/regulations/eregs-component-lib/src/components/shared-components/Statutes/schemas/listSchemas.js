const titleSelectorList = {
    ssa: {
        name: "Social Security Act",
        titles: [
            {
                title: "11",
                titleRoman: "XI",
            },
            {
                title: "18",
                titleRoman: "XVIII",
            },
            {
                title: "19",
                titleRoman: "XIX",
            },
            {
                title: "21",
                titleRoman: "XXI",
            },
        ],
    },
};

export { titleSelectorList };
