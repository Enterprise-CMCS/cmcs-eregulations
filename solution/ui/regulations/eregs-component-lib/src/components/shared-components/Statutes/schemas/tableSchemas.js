import {
    houseGovUrl,
    ssaGovUrl,
    statuteCompilationUrl,
    usCodeUrl,
} from "../utils/urlMethods.js";
import { getDateLabel } from "../utils/dateMethods.js";

// placeholder for future schema
const acaSchema = [
    {
        header: {
            title: "Primary Column",
            primary: true,
        },
        body: {
            title: "ACA Title",
            label: "ACA Label",
            name: "ACA Name",
            primary: true,
        },
    },
    {
        header: {
            title: "Secondary Column",
            secondary: true,
            subtitles: ["Subtitle One", "Subtitle Two"],
        },
        body: {
            url: "Link href",
            text: "Link text",
            type: "external", // or "pdf"
            secondary: true,
        },
    },
];

const ssaSchema = [
    {
        header: {
            title: "Statute Citation",
            primary: true,
        },
        body: {
            title: (statute) => `SSA Section ${statute.section}`,
            label: (statute) => `${statute.title} U.S.C. ${statute.usc}`,
            name: (statute) => `${statute.name}`,
            primary: true,
        },
    },
    {
        header: {
            testId: "house-gov-link",
            title: "US Code House.gov",
            secondary: true,
            subtitles: [
                () => "Web Page",
                (columnDates) =>
                    getDateLabel(columnDates?.us_code_house_gov ?? {}),
            ],
        },
        body: {
            url: (statute) => houseGovUrl(statute),
            text: (statute) => `${statute.usc}`,
            type: "external",
            secondary: true,
        },
    },
    {
        header: {
            testId: "statute-compilation-link",
            title: "Statute Compilation",
            secondary: true,
            subtitles: [
                () => "PDF Document",
                (columnDates) =>
                    getDateLabel(columnDates?.statute_compilation ?? {}),
            ],
        },
        body: {
            url: (statute) => statuteCompilationUrl(statute),
            text: (statute) => `Title ${statute.statute_title_roman}`,
            type: "pdf",
            secondary: true,
        },
    },
    {
        header: {
            testId: "us-code-annual-link",
            title: "US Code Annual",
            secondary: true,
            subtitles: [
                () => "PDF Document",
                (columnDates) =>
                    getDateLabel(columnDates?.us_code_annual ?? {}),
            ],
        },
        body: {
            url: (statute) => usCodeUrl(statute),
            text: (statute) => `${statute.usc}`,
            type: "pdf",
            secondary: true,
        },
    },
    {
        header: {
            testId: "ssa-gov-link",
            title: "SSA.gov Compilation",
            secondary: true,
            subtitles: [
                () => "Web Page",
                (columnDates) =>
                    getDateLabel(columnDates?.ssa_gov_compilation ?? {}),
            ],
        },
        body: {
            url: (statute) => ssaGovUrl(statute),
            text: (statute) => `${statute.section}`,
            type: "external",
            secondary: true,
        },
    },
];

export { acaSchema, ssaSchema };
