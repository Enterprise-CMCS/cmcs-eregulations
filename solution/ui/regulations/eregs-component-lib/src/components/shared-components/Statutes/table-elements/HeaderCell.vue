<script setup>
import { DISPLAY_TYPES } from "../utils/enums";

const props = defineProps({
    cellData: {
        type: Object,
        required: true,
    },
    displayType: {
        validator: (value) => DISPLAY_TYPES.includes(value),
        required: false,
        default: "table",
    },
    columnDates: {
        type: Object,
        required: false,
        default: () => {},
    },
});
</script>

<template>
    <th
        class="row__cell row__cell--header"
        :class="{
            'row__cell--primary': cellData.primary,
            'row__cell--secondary': cellData.secondary,
        }"
    >
        <div class="cell__title">
            {{ props.cellData.title }}
        </div>
        <template v-if="cellData.subtitles">
            <div
                v-for="(subtitle, i) in cellData.subtitles"
                :key="`${props.displayType}-subtitle-${i}`"
                class="cell__subtitle"
                :data-testid="`${props.cellData.testId}-subtitle-${i}`"
            >
                {{ subtitle(columnDates) }}
            </div>
        </template>
    </th>
</template>
