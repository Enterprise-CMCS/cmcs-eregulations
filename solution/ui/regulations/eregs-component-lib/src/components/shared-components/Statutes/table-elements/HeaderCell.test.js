import flushPromises from "flush-promises";
import { render } from "@testing-library/vue";
import { describe, it, expect } from "vitest";

import statuteDatesFixture from "cypress/fixtures/statute-dates.json";

import { ssaSchema } from "../schemas/tableSchemas";

import HeaderCell from "./HeaderCell.vue";

describe("Statute Table Header Cell", () => {
    describe("SSA table header", () => {
        const secondaryCells = ssaSchema.filter(
            (column) => column.header.secondary
        );

        secondaryCells.forEach((column, index) => {
            it(`Creates a snapshot of header cell for column ${index + 1} without dates`, async () => {
                const wrapper = render(HeaderCell, {
                    props: {
                        cellData: column.header,
                        displayType: "table",
                    },
                });
                await flushPromises();

                const dateCell = wrapper.getByTestId(
                    `${column.header.testId}-subtitle-1`
                ).textContent;
                expect(dateCell.trim()).toEqual("");

                expect(wrapper).toMatchSnapshot();
            });

            it(`Creates a snapshot of header cell for column ${index + 1} with dates`, async () => {
                const wrapper = render(HeaderCell, {
                    props: {
                        cellData: column.header,
                        displayType: "table",
                        columnDates: statuteDatesFixture,
                    },
                });
                await flushPromises();

                const dateCell = wrapper.getByTestId(
                    `${column.header.testId}-subtitle-1`
                ).textContent;
                expect(dateCell.trim()).toEqual("effective Aug 2023");

                expect(wrapper).toMatchSnapshot();
            });
        });
    });
});
