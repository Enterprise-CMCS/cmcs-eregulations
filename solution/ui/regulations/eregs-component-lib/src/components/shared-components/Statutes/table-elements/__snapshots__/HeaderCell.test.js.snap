// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Statute Table Header Cell > SSA table header > Creates a snapshot of header cell for column 1 1`] = `
{
  "baseElement": <body>
    <div>
      <th
        class="row__cell row__cell--header row__cell--primary"
      >
        <div
          class="cell__title"
        >
           Statute Citation 
        </div>
        <!---->
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="row__cell row__cell--header row__cell--primary"
    >
      <div
        class="cell__title"
      >
         Statute Citation 
      </div>
      <!---->
    </th>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Header Cell > SSA table header > Creates a snapshot of header cell for column 2 1`] = `
{
  "baseElement": <body>
    <div>
      <th
        class="row__cell row__cell--header row__cell--secondary"
      >
        <div
          class="cell__title"
        >
           US Code House.gov 
        </div>
        <div
          class="cell__subtitle"
        >
           Web Page 
        </div>
        <div
          class="cell__subtitle"
        >
           Effective Jul 2023 
        </div>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="row__cell row__cell--header row__cell--secondary"
    >
      <div
        class="cell__title"
      >
         US Code House.gov 
      </div>
      <div
        class="cell__subtitle"
      >
         Web Page 
      </div>
      <div
        class="cell__subtitle"
      >
         Effective Jul 2023 
      </div>
    </th>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Header Cell > SSA table header > Creates a snapshot of header cell for column 3 1`] = `
{
  "baseElement": <body>
    <div>
      <th
        class="row__cell row__cell--header row__cell--secondary"
      >
        <div
          class="cell__title"
        >
           Statute Compilation 
        </div>
        <div
          class="cell__subtitle"
        >
           PDF Document 
        </div>
        <div
          class="cell__subtitle"
        >
           Amended Dec 2022 
        </div>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="row__cell row__cell--header row__cell--secondary"
    >
      <div
        class="cell__title"
      >
         Statute Compilation 
      </div>
      <div
        class="cell__subtitle"
      >
         PDF Document 
      </div>
      <div
        class="cell__subtitle"
      >
         Amended Dec 2022 
      </div>
    </th>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Header Cell > SSA table header > Creates a snapshot of header cell for column 4 1`] = `
{
  "baseElement": <body>
    <div>
      <th
        class="row__cell row__cell--header row__cell--secondary"
      >
        <div
          class="cell__title"
        >
           US Code Annual 
        </div>
        <div
          class="cell__subtitle"
        >
           PDF Document 
        </div>
        <div
          class="cell__subtitle"
        >
           Effective Jan 2022 
        </div>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="row__cell row__cell--header row__cell--secondary"
    >
      <div
        class="cell__title"
      >
         US Code Annual 
      </div>
      <div
        class="cell__subtitle"
      >
         PDF Document 
      </div>
      <div
        class="cell__subtitle"
      >
         Effective Jan 2022 
      </div>
    </th>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Header Cell > SSA table header > Creates a snapshot of header cell for column 5 1`] = `
{
  "baseElement": <body>
    <div>
      <th
        class="row__cell row__cell--header row__cell--secondary"
      >
        <div
          class="cell__title"
        >
           SSA.gov Compilation 
        </div>
        <div
          class="cell__subtitle"
        >
           Web Page 
        </div>
        <div
          class="cell__subtitle"
        >
           Amended Dec 2019 
        </div>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="row__cell row__cell--header row__cell--secondary"
    >
      <div
        class="cell__title"
      >
         SSA.gov Compilation 
      </div>
      <div
        class="cell__subtitle"
      >
         Web Page 
      </div>
      <div
        class="cell__subtitle"
      >
         Amended Dec 2019 
      </div>
    </th>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
